### PR TITLE
Execute imports and exports by name

### PR DESCRIPTION
**Pull request by blankse**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 

I needs to execute the export and import command by name

### CODE REVIEW

File: src/ImportDefinitionsBundle/Command/ExportCommand.php [hunk 1]
```diff
@@ -87,7 +88,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $eventDispatcher = $this->eventDispatcher;
 
         $params = json_decode($input->getOption('params'), true);
-        $definition = $this->repository->find($input->getOption('definition'));
+        try {
+            $definition = $this->repository->find($input->getOption('definition'));
+        } catch (\InvalidArgumentException $e) {
+            $definition = ExportDefinition::getByName($input->getOption('definition'));
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Command/ExportCommand.php]: make use of the repository here

File: src/ImportDefinitionsBundle/Command/ImportCommand.php [hunk 2]
```diff
@@ -90,7 +91,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $eventDispatcher = $this->eventDispatcher;
 
         $params = $input->getOption('params');
-        $definition = $this->repository->find($input->getOption('definition'));
+        try {
+            $definition = $this->repository->find($input->getOption('definition'));
+        } catch (\InvalidArgumentException $e) {
+            $definition = ImportDefinition::getByName($input->getOption('definition'));
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Command/ImportCommand.php]: make use of the repository here

File: src/ImportDefinitionsBundle/Command/ExportCommand.php (hunk 1 above)
Comment (blankse) [reply src/ImportDefinitionsBundle/Command/ExportCommand.php]: I tried this. But is impossible in PhpArrayTable.

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Command/ExportCommand.php]: why? it works with the default repository as well.

Comment (blankse) [reply src/ImportDefinitionsBundle/Command/ExportCommand.php]:
```
        try {
            $definition = $this->repository->find($input->getOption('definition'));
        } catch (\InvalidArgumentException $e) {
            $definition = $this->repository->findOneBy([
                'name' => $input->getOption('definition')
            ]);
        }
```
Error:
`Call to undefined method addConditionParam in class ImportDefinitionsBundle\Model\ExportDefinition\Listing`

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Command/ExportCommand.php]: True, you shouldn't use that.

What I mean is, you can create the Repository class, inherit from the default one and change the configuration in Configuration.php to use the new Repository Class. In there you have access to the className and you can do what you are doing in the model

**Comment by blankse** [comment]
@dpfaffenbauer Is it so okay?

**Comment by dpfaffenbauer** [comment]
better :) thanks

File: src/ImportDefinitionsBundle/Repository.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class Repository extends PimcoreRepository
+{
+    /**
+     * @param string $name
+     * @return object|null
+     */
+    public function findByName($name)
+    {
+        $class = $this->metadata->getClass('model');
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Repository.php]:
```
$class = $this->metadata->getClass('model');
$definitionEntry = new $className();
$definitionEntry->getDao()->getByName($name);
return $definitionEntry;
```

and remove it from the model. The model should never know about its storage.

Comment (blankse) [reply src/ImportDefinitionsBundle/Repository.php]: I changed it

**Comment by dpfaffenbauer** [comment]
thanks :)

**Pull request merged**